OMCompiler/SimulationRuntime/c/util/simulation_options.c

```diff
@@ -568,6 +568,125 @@ const char *FLAG_DETAILED_DESC[FLAG_MAX+1] = {
   "FLAG_MAX"
 };
 
+const flag_repeat_policy FLAG_REPEAT_POLICIES[FLAG_MAX] = {
```

Comment:
This looks a bit fragile to me. What if someone adds one more flag and forgets to update this array?

Reply:
Also, this information should probably end up in the automatically generated documentation, so we know what is the policy for each flag without looking at the source code.

With some introductory explanation [here](https://openmodelica.org/doc/OpenModelicaUsersGuide/latest/simulationflags.html#cruntime-simflags)

Reply:
Last, but not least, it is not clear to me how this combines with the __OpenModelica_simulationFlags annotation. Are they applied first, and then whatever is specified on the command line is applied afterwards, or is it the other way round? This should be explained [here](https://openmodelica.org/doc/OpenModelicaUsersGuide/latest/omedit.html#openmodelica-simulationflags-annotation)

Reply:
> This looks a bit fragile to me. What if someone adds one more flag and forgets to update this array?

This is how the flag management works right now. If you add a flag you need to update things in all the proper places. However, it is not very complicated. You just have to be thorough. It can usually be done by _search and duplicate_ kind of operation starting from an existing flag.

Maybe the design can be improved if really needed. However, for me, it seems okay. It is easy to understand and modify, and it does the job just fine. Flags also do not change frequently. I think the overall design is mostly a consequence of the language, which is C.

Reply:
> Also, this information should probably end up in the automatically generated documentation, so we know what is the policy for each flag without looking at the source code.
> 
> With some introductory explanation [here](https://openmodelica.org/doc/OpenModelicaUsersGuide/latest/simulationflags.html#cruntime-simflags)

Yes. We will need to add information on which flags are allowed to be repeated. For those not mentioned it will be an error and we can clarify that there.

Reply:
> Last, but not least, it is not clear to me how this combines with the __OpenModelica_simulationFlags annotation. Are they applied first, and then whatever is specified on the command line is applied afterwards, or is it the other way round? This should be explained [here](https://openmodelica.org/doc/OpenModelicaUsersGuide/latest/omedit.html#openmodelica-simulationflags-annotation)

It depends on which policy we pick for each flag. **If the policy is **replace** then whichever came last will be kept. This  means anything repeated will override the annotation flags**. **If the policy is *ignore* then the first one is kept, which means the ones in the annotation will be final and can not be overridden**. 

For example for `lv` what is **specified outside will override** what is in the annotations. Because the policy is **replace at the moment** and the annotation options came first.  If you want it to flip it around and always keep the annotation options we just switch the policy to "ignore" and **recompile OpenModelica** (the policies can not be changed by the users, they are for us, developers and we decide beforehand what the rule is for each flag).

Reply:
Of course the actual effect will depend on the individual flag policy. My question was about the order, which comes first: the annotations or the flags passed to the executable at runtime? This should be stated in the documentation.

Reply:
As of right now the annotations will come first as far as I can tell. I will add it to the documentation for `lv`.